File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& nums) {
        int n=nums.size();
        int i=0,j=n-1,ans=INT_MIN;
        while(i<j){
            int height=min(nums[i],nums[j]);
            int width=j-i;
            ans=max(ans,height*width);
            if(nums[i]<nums[j])i++;
            else{
                j--;
            }
        }
        return ans;
        
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    const n=height.length;
    let ans=0;
    let i=0,j=n-1;
    while(i<=j){
        const h=Math.min(height[i],height[j]);
        const w=j-i;
        ans=Math.max(ans,h*w);
        if(height[i]<height[j]){
            i++;
        }
        else{
            j--;
        }
    }
    return ans;
};